adapter/internal/ga/client.go

```diff
@@ -288,13 +290,98 @@ func addAPIToChannel(resp *discovery.DiscoveryResponse) {
 			IsDeployEvent:    false,
 			OrganizationUUID: gaAPI.OrganizationUUID,
 			RevisionUUID:     gaAPI.RevisionUUID,
+			DeployedEnv:      gaAPI.DeployedEnv,
 		}
 		GAAPIChannel <- event
 		delete(apiRevisionMap, apiEntry)
 		logger.LoggerGA.Infof("API Undeploy event is added to the channel. : %s", apiEntry)
 	}
 }
 
+func addAPIWithEnvToChannel(resp *discovery.DiscoveryResponse) {
+
+	// removedAPIEnvMap is used To keep track of the APIs needs to be deleted.
+	var removedAPIEnvMap = make(map[string]map[string]*ga_model.Api)
+
+	if !isFirstResponse {
+		for k, v := range apiEnvRevisionMap {
+			cp := make(map[string]*ga_model.Api)
+			for l, w := range v {
+				cp[l] = w
+			}
+			removedAPIEnvMap[k] = cp
+		}
+	}
+
+	var startupAPIEventArray []*APIEvent
+	// Even if there are no resources available within ga, an empty but non-nil array would be set
+	startupAPIEventArray = make([]*APIEvent, 0)
+	for _, res := range resp.Resources {
+		api := &ga_model.Api{}
+		err := ptypes.UnmarshalAny(res, api)
+
+		if err != nil {
+			logger.LoggerGA.Errorf("Error while unmarshalling: %s\n", err.Error())
+			continue
+		}
+
+		currentGAAPI, apiFound := apiEnvRevisionMap[api.ApiUUID][api.DeployedEnv]
```

Comment:
Let's assume that it is a newly created API, then it would end up with NPE

Reply:
where apiEnvRevisionMap[api.ApiUUID] == nil

Reply:
No, there is no NPE here. What happens is `apiFound` becomes `false` and `currentGAAPI` becomes nil. I have verified this scenario in locally debugging.

Reply:
Oh it seems golang handles it own its own.